### PR TITLE
Enchancement system token manager pallet

### PR DESCRIPTION
**Pull request by sweatpotato13**
### Summary
**See #40**
* Apply https://github.com/InfraBlockchain/infra-substrate/commit/9ecdc892925f53467592f654e922b0bf194707cc changes
* Change name `SystemTokenExchangeRate` to `SystemTokenProperties(SystemTokenId, SystemTokenProperty)`
* Add timestamp in struct `SystemTokenProperty`
* Add timestamp in struct `WrappedSystemTokenProperty`
* Add Storage `ParaIdsBySystemToken` that Reverse storage of `SystemTokenOnParachainByParaId`
* And some refactoring things

and so on...

### Describe your changes
```rust
#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Default, TypeInfo, MaxEncodedLen)]
pub struct SystemTokenProperty {
	/// weight of this system token
	pub weight: SystemTokenWeight,
	/// epoch time of this system token registered
	pub created_at: u128,
}

#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Default, TypeInfo, MaxEncodedLen)]
pub struct WrappedSystemTokenProperty {
	/// epoch time of this system token registered
	pub created_at: u128,
}
```

```rust
	#[pallet::storage]
	#[pallet::getter(fn system_token_properties)]
	/// List for original system token and metadata.
	pub(super) type SystemTokenProperties<T: Config> =
		StorageMap<_, Twox64Concat, SystemTokenId, SystemTokenProperty, OptionQuery>;

	#[pallet::storage]
	#[pallet::getter(fn wrapped_system_token_properties)]
	/// List for wrapped system token and metadata.
	pub(super) type WrappedSystemTokenProperties<T: Config> =
		StorageMap<_, Twox64Concat, WrappedSystemTokenId, WrappedSystemTokenProperty, OptionQuery>

	#[pallet::storage]
	#[pallet::getter(fn para_ids_by_system_token)]
	/// List of parachain ids using system token
	pub(super) type ParaIdsBySystemToken<T: Config> = StorageMap<
		_,
		Twox64Concat,
		SystemTokenId,
		BoundedVec<ParaId, T::MaxWrappedSystemToken>,
		OptionQuery,
	>;
```

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] Well documented code
- [x] `cargo clippy`
- [x] `cargo-nightly fmt`

### CODE REVIEW

File: runtime/parachains/src/system_token_manager.rs [hunk 1]
```diff
@@ -269,33 +319,29 @@ pub mod pallet {
 				!SystemTokenOnParachain::<T>::contains_key(&wrapped_system_token),
 				Error::<T>::WrappedSystemTokenAlreadyRegistered
 			);
+			ensure!(
+				!ParaIdsBySystemToken::<T>::get(&system_token_id)
+					.unwrap()
```
Comment (cocoyoon) [inline runtime/parachains/src/system_token_manager.rs]: unwrap() 위험하지 않을까요?

File: runtime/parachains/src/system_token_manager.rs [hunk 2]
```diff
@@ -269,33 +319,29 @@ pub mod pallet {
...
+					.contains(&wrapped_system_token.clone().para_id),
+				Error::<T>::SameSystemTokenAlreadyRegistered
```
Comment (energyGiver) [inline runtime/parachains/src/system_token_manager.rs:326]: I believe using match would be a better option than .unwrap to handle the result of ::get(), as the latter can cause a panic.

In the system_token_manager, I hope the rest of the unwrap calls can be replaced with match or other methods!

**Review by energyGiver — CHANGES_REQUESTED** [review]
I believe using match would be a better option than .unwrap to handle the result of ::get(), as the latter can cause a panic.

In the system_token_manager, I hope the rest of the unwrap calls can be replaced with match or other methods!

File: runtime/parachains/src/system_token_manager.rs (hunk 2 above)
Comment (sweatpotato13) [inline runtime/parachains/src/system_token_manager.rs:340]: Change to this
```
			let para_ids = {
				match ParaIdsBySystemToken::<T>::get(&system_token_id) {
					Some(para_ids) => para_ids,
					None => Default::default(),
				}
			};

			ensure!(
				!para_ids.contains(&wrapped_system_token.clone().para_id),
				Error::<T>::SameSystemTokenAlreadyRegistered
			);

```

Comment (cocoyoon) [reply runtime/parachains/src/system_token_manager.rs:340]: Or you can simply use 
```rust 
let para_ids = ParaIdsBySystemToken::<T>::get().map_or(Default::default(), |_| para_ids);
```

Comment (sweatpotato13) [reply runtime/parachains/src/system_token_manager.rs:340]: Fix to
```
let para_ids = ParaIdsBySystemToken::<T>::get(&system_token_id).map_or(Default::default(), |para_id| para_id);
```

**Review by cocoyoon — APPROVED** [review]

**Review by energyGiver — APPROVED** [review]

**Pull request merged**